master/docs/developer/data.rst

```diff
@@ -285,7 +285,7 @@ In ``master/buildbot/data/pubs.py``, create a subclass of :py:class:`ResourceTyp
 
     .. py:attribute:: eventPathPatterns
 
-        :type: str
+        :type: list of strings
```

Comment:
Also add a note about this change to 5.0 upgrade documentation at master/docs/manual/upgrading/5.0-upgrade.rst